ext/js/data-display.js

```diff
@@ -108,18 +123,134 @@ function DataDisplay(loads, rides) {
     return { bus: busRides, train: trainRides, unknown: unknownRides }
   }
   
-  function overallRideTypeCounts() {
+  this.overallRideTypeCounts = function() {
     return this.rideTypeCountsForRides(this.rides);
   }
 
   //TIME & DATE
   
+  //Month
+  
+  this.monthsInPlay = function() {
+    if (this.monthsPlay != null) {
+      return this.monthsPlay;
+    }
+    
+    this.monthsPlay = [];
+    
+    var firstMonth = this.oldestRideDate().month()
+    var lastMonth = this.newestRideDate().month();
+    
+    if (firstMonth > lastMonth) {
+      lastMonth += 12; //
+    }
+    
+    for (var i = firstMonth; i < lastMonth; i++) {
+      this.monthsPlay.push(i);
+    }
+    
+    return this.monthsPlay;
+  }
+  
+  this.allMonthLabels = function() {
+    //If we've already figured this out, just return it. 
+    if (this.monthNamesInPlay != null) {
+      return this.monthNamesInPlay
+    }
+    
+    var months = this.monthsInPlay();
+    this.monthNamesInPlay = [];
+    
+    for (var i = 0; i < months.length; i++) {
+      var month = months[i];      
+      var monthIndex = month % 12; 
+      
+      switch (monthIndex) {
+        case 0:
+          this.monthNamesInPlay.push("Jan");
+          break;
+        case 1:
+          this.monthNamesInPlay.push("Feb");
+          break;
+        case 2:
+          this.monthNamesInPlay.push("Mar");
+          break;
+        case 3:
+          this.monthNamesInPlay.push("Apr");
+          break;
+        case 4:
+          this.monthNamesInPlay.push("May");
+          break;
+        case 5:
+          this.monthNamesInPlay.push("Jun");
+          break;
+        case 6:
+          this.monthNamesInPlay.push("Jul");
+          break;
+        case 7:
+          this.monthNamesInPlay.push("Aug");
+          break;
+        case 8:
+          this.monthNamesInPlay.push("Sep");
+          break;
+        case 9:
+          this.monthNamesInPlay.push("Oct");
+          break;
+        case 10:
+          this.monthNamesInPlay.push("Nov");
+          break;
+        case 11:
+          this.monthNamesInPlay.push("Dec");
+          break;
+        default:
+          this.monthNamesInPlay.push("Smarch");
+          break;  
+      }
```

Comment:
You can make this switch substantially more efficient by just using an array to get the months by referencing the position in the array